hbase_sql/src/main/java/com/splicemachine/olap/OlapServerSubmitter.java

```diff
@@ -209,6 +211,17 @@ public void run() {
                         if (!appState.equals(YarnApplicationState.RUNNING)) {
                             reportDiagnostics("YARN state for application is " + appState +". Not enough YARN resources?");
                         }
+
+                        if (!watcherSet && appState.equals(YarnApplicationState.RUNNING)) {
```

Comment:
The `watcherSet` field disables checking YARN for the OlapServer status, if it dies how do we restart it?

Reply:
the watcherSet field here only make sure we only set one watcher for a running olap server. The check for Olapserver status to restart is in the outer while loop

Reply:
I see, thanks, I got confused!